### DISCUSSION TITLE
Setting port label opacity

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2199] [Q&A]
### Introduction

I am trying to implement a zoom related feature in jointjs+, where initially the port label's opacity of my custom card is 0. While zooming in I want to update the opacity. Tried different way but not getting any solution. could you guide me?
Tried this below:
```js
const changePortLabelOpacity = (opacity1:number) => {
      graph.getElements().forEach((element) => {
        //element.prop(''attrs/portLabel'/opacity',opacity)
       // element.prop('attrs/portLabel', { opacity: opacity1 }, { rewrite: true });
        const ports = element.getPorts();
        ports.forEach((port) => {
          // const portLabel = port.attr('portLabel');
          // if (portLabel) {
          //   portLabel.opacity = opacity;
          // }
        });
      });
    };
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
It might be easier to add a CSS class when the zoom level hits a certain threshold, and hide the ports with CSS only.

CSS:
```css
.zoom-in .joint-port {
    opacity: 0;
}
```

JS:
```js
paper.on('scale', (scale) => {
    paper.el.classList.toggle('zoom-in', scale > 1);
});
```

**Reply by ROYopedia** [reply]
@kumilingus It is doing the exact opposite what I want. By default opacity of the port label should be 0 and while zooming(Scale) it should be 1. 
Tried with your example to achieve but it seems it is not working.
Also, With your suggestion port label along with port is going away.

**Reply by kumilingus** [reply]
CSS:
```css
/* `circle` sub element or whatever SVG shape you use as port */
.zoom-out .joint-port circle {
    opacity: 0;
}
```

JS:
```js
//  reversed logic to `zoom-in` from the previous comment
paper.on('scale', (scale) => {
    paper.el.classList.toggle('zoom-out', scale < 1);
});

paper.el.classList.add('zoom-out');
```
